code-challenges.test.js

```diff
@@ -13,34 +13,124 @@
+  })
+})
+// 
 // b) Create the function that makes the test pass.
-
+// Pseudocode:
+// 
+// Input - array of objects with 2 key:value pairs
+// Output - array with a sentence about each person with their name capitalized
+// Process:
+// function to capitalize each word in a string
+// using the built in method .split(' ') using a space inside the single quotations as the delimiter for the new array
+// using .map(word => ...) to create a new array populated with the results of calling a provided function on every element in the calling array, the function
+// calls each element in the array in order
+// 'word.chatAt(0).toUpperCase()' -> the .charAt method returns a new string with the substring located at index 0 , the .toUpperCase method converts a string
+// to uppercase letters. 
+// '+' -> concatenating the elements
+// 'word.slice(1)' -> the .slice method extracts a section of a string and returns it as a new string, without modifying the original string
+// '.join(' ') -> using the .join method to create and return a new string concatenating all of the elements of an array, separated by the string separator( space in this case)
+// 
+const capitalizedNames = (str) => str.split(' ').map(word => word.charAt(0).toUpperCase() + word.slice(1)).join(' ');
+  
+  const objArray = (chars) => { 
+    return chars.map(char => `${capitalizedNames(char.name)} is ${char.occupation}.`)
+  }
+// 
+// ReferenceError: objArray is not defined
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+// objArray
+// ✓ returns an array with a sentence about each person with their name capitalized (4 ms)
+// 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
-
+// 
 // a) Create a test with an expect statement using the variables provided.
+// 
+describe("onlyRemainders", () => {
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false];
+  const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true];
+  it("returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    expect(onlyRemainders(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(onlyRemainders(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
+// 
+// b) Create the function that makes the test pass.
+// 
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+// Pseudocode:
+// 
+// Input - array of objects with 2 key:value pairs
+// Output - array with a sentence about each person with their name capitalized
+// Process:
+// define a function named 'onlyRemainders'
+// create a variable named 'newArray' with an empty array
+// iterate through the function using a for loop
+// use a conditional statement with the typeof method to separate the numbers in the array
+// then push the product of the array integers by index divided by the modulo of 3 into the new array
```

Comment:
Good understanding, I would consider a refactor not using the newArray variable at all, also using an arrow function.